OutOfSchool/OutOfSchool.DataAccess/Repository/CodeficatorRepository.cs

```diff
@@ -34,13 +34,17 @@ public CodeficatorRepository(OutOfSchoolDbContext dbContext)
     /// <inheritdoc/>
     public async Task<List<CodeficatorAddressDto>> GetFullAddressesByPartOfName(string namePart, string categories = default)
     {
+        int cityAmountIfNamePartIsEmpty = 10;
+
         var query = from e in db.CATOTTGs
                      from p in db.CATOTTGs.Where(x1 => e.ParentId == x1.Id).DefaultIfEmpty()
                      from pp in db.CATOTTGs.Where(x2 => p.ParentId == x2.Id).DefaultIfEmpty()
                      from ppp in db.CATOTTGs.Where(x3 => pp.ParentId == x3.Id).DefaultIfEmpty()
                      from pppp in db.CATOTTGs.Where(x4 => ppp.ParentId == x4.Id).DefaultIfEmpty()
-                     where ((CodeficatorCategory.Level4.Name.Contains(e.Category) && e.Name.StartsWith(namePart)) || (e.Category == CodeficatorCategory.CityDistrict.Name && p.Name.StartsWith(namePart))) && categories.Contains(e.Category)
-                     select new CodeficatorAddressDto
+                     where string.IsNullOrEmpty(namePart)
```

Comment:
"Mad" query becomes even worser. Though I had no suggestions right now, I wish next visitor to this code refactor it.